### PR TITLE
OKTA-719549 | Create Banner Announcement for DOC

### PR DESCRIPTION
**Pull request by madhurishivakumar-okta**
## Description:
- **What's changed?**  This PR implements the new banner on top of the header.
- **Is this PR related to a Monolith release?**  No

### Resolves:

* [OKTA-719549](https://oktainc.atlassian.net/browse/OKTA-719549)

### Changes:

1. Dismissible Feature:
   - Prop: `dismissible`
   - Functionality: Controls the display of dismiss button on the banner
   - Usage: Set to `default: false` , pass `true` to show close icon
   - Current Implementation: For the current content, dismissible is set to false as per the Jira ticket
2. Persistent Banner state:  
   - Prop: `bannerId`
   - Functionality: Uses bannerId as a unique string to store the dismissal state in localStorage to ensure the banner remains dismissed across sessions.

Screenshot 2024-05-21 at 1 44 34 PM

https://github.com/okta/okta-developer-docs/assets/158277016/fe7aabb2-8980-40e4-ad90-62d6ba8bf813

### CODE REVIEW

**Review by emikhasyak-okta — COMMENTED** [review]
The banner looks nice!
- The navigation is a bit off on errors page and on other pages too, but sometimes less noticeable.
- Can user dismiss banner? if user has completed survey or don't want to complete it at all.

**Comment by madhurishivakumar-okta** [comment]
> The banner looks nice!
> 
> * The navigation is a bit off on errors page and on other pages too, but sometimes less noticeable.
- I will fix this.

> * Can user dismiss banner? if user has completed survey or don't want to complete it at all.
- For this banner, @kimmaida-okta mentioned that we will retain the banner for a couple of weeks without allowing users to dismiss and then push a separate build to remove it. 
However, the ticket's acceptance criteria also mentions to extend the capability of dismissing the banner, which I have implemented.

File: packages/@okta/vuepress-theme-prose/layouts/Layout.vue [hunk 1]
```diff
@@ -195,6 +213,12 @@ export default {
     this.$on("toggle-tree-nav", event => {
       that.appContext.isTreeNavMobileOpen = event.treeNavOpen;
     });
+    
+    // Delay height adjustment on mount to ensure the header element is fully rendered.
+    setTimeout(() => {
```
Comment (emikhasyak-okta) [inline packages/@okta/vuepress-theme-prose/layouts/Layout.vue:218]: I guess we don't have a good way to avoid `setTimeout` approach...

**Review by emikhasyak-okta — APPROVED** [review]
Great job! 🚀

File: packages/@okta/vuepress-theme-prose/layouts/Layout.vue (hunk 1 above)
Comment (madhurishivakumar-okta) [reply packages/@okta/vuepress-theme-prose/layouts/Layout.vue:218]: Yes, especially when calculating height in javascript.

**Pull request merged**